main2.cpp

```diff
@@ -1056,23 +1056,31 @@ void debug_plot_markmap() {
 	while(true);
 }
 
+/* Return true when FPU is available */
+bool cpu_enable_fpu(void) 
+{
+	uint32_t fpuEnable = 0b1111 << 20;
+	if((SCB_CPACR & fpuEnable) != fpuEnable) {
+		SCB_CPACR |= fpuEnable;
+		if((SCB_CPACR & fpuEnable) != fpuEnable) {
+			return false;
+		} 
+	}
+	return true;
+}
 
 int main(void) {
-	uint32_t fpuEnable = 0b1111 << 20;
 	bool shouldShowDmesg = false;
 
 #ifndef GD32F3_NOFPU
-	if((SCB_CPACR & fpuEnable) != fpuEnable) {
-		SCB_CPACR |= fpuEnable;
-		if((SCB_CPACR & fpuEnable) != fpuEnable) {
-			// printk1() does not invoke printf() and does not use fpu
+	if(cpu_has_fpu()) {
```

Comment:
is this supposed to be cpu_enable_fpu()?

Reply:
Yes, I renamed the function since it not only detects but also enables. Sorry.

Reply:
I added a new commit. Fix it.